frontend/packages/component-library/src/tooltip/WithTooltip.tsx

```diff
@@ -85,45 +98,77 @@ const WithTooltip: React.FunctionComponent<WithTooltipProps> = ({
     updateTooltipStyles,
   ]);
 
+  // Effect to handle the Escape key to close the tooltip
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && !event.defaultPrevented) {
+        handleHideTooltip();
+        event.preventDefault(); // Prevent other listeners from handling the event
+      }
+    };
+
+    if (showTooltip) {
+      window.addEventListener('keydown', handleKeyDown);
+    }
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [showTooltip]);
+
   const tooltipStyleProps: React.CSSProperties = {
     visibility: showTooltip ? 'visible' : 'hidden',
     ...tooltipStyles,
   };
 
-  // Check if children is a valid React element and clone it with ariaDescribedBy attribute
-  // and additional event handlers to make sure the tooltip is displayed correctly
-  const componentToWrap =
-    isValidElement<HTMLAttributes<HTMLElement>>(children) &&
-    cloneElement(children, {
-      'aria-describedby': tooltipProps.tooltipId,
-      onFocus: (event: React.FocusEvent<HTMLElement>) => {
-        handleShowTooltip(true, event);
+  // Wrap children in a container with event handlers and additional padding
+  const containerProps = {
+    'aria-describedby': tooltipProps.tooltipId,
+    onFocus: (event: React.FocusEvent<HTMLElement>) => {
+      handleShowTooltip(true, event);
+      if (isValidElement(children)) {
         children.props.onFocus?.(event);
-      },
-      onBlur: (event: React.FocusEvent<HTMLElement, Element>) => {
-        handleShowTooltip(false, event);
+      }
+    },
+    onBlur: (event: React.FocusEvent<HTMLElement>) => {
+      handleHideTooltip();
+      if (isValidElement(children)) {
         children.props.onBlur?.(event);
-      },
-      onMouseEnter: (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
-        handleShowTooltip(true, event);
+      }
+    },
+    onMouseEnter: (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+      handleShowTooltip(true, event);
+      if (isValidElement(children)) {
         children.props.onMouseEnter?.(event);
-      },
-      onMouseLeave: (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
-        handleShowTooltip(false, event);
+      }
+    },
+    onMouseLeave: (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+      handleHideTooltip();
+      if (isValidElement(children)) {
         children.props.onMouseLeave?.(event);
-      },
-    });
+      }
+    },
+    tabIndex: 0, // Make the container focusable
+  };
 
   return (
     <TooltipOverlay className={tooltipOverlayClassName}>
-      {componentToWrap}
+      <div {...containerProps}>{children}</div>
       {showTooltip &&
         createPortal(
-          <Tooltip
-            {...tooltipProps}
-            ref={tooltipRef}
-            style={tooltipStyleProps}
-          />,
+          // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+          <div
```

Comment:
@levadadenys I'm curious if you have in mind a solution aside from wrapping the Tooltip in a div so it remains open on hover. This is causing an issue in the Tab component because an <ul> isn't supposed to contain divs, but I've been banging my head against finding another way to do it